default/python/AI/FreeOrionAI.py

```diff
@@ -257,6 +257,21 @@ def generateOrders():  # pylint: disable=invalid-name
             error("Exception %s while trying doneTurn() on eliminated empire" % e, exc_info=True)
         return
 
+    universe = fo.getUniverse()
+    if not any(planet and planet.speciesName and planet.ownedBy(empire.empireID)
+               for planet in map(universe.getPlanet, universe.planetIDs)):
+        debug("This empire has no colonies left. Surrendering.")
+        if not foAIstate.surrendered:
+            foAIstate.surrender()
```

Comment:
What about colony ships?

Reply:
Could in principle be added but I think the AI would already be so weak at that point, that it wouldn't matter in terms of posing a challenge.

Dragging the game out by having it surrender too late can make the last part of the game feel like a chore and rather unfun. Making the AI surrender too quickly after the first losses, however, is unsatisfying for the player as well.

Considering only the colonized planets as surrender condition feels like a good starting point with trivial implementation compared to keeping track of how quickly it loses planets etc and trying to deduce from that if it should surrender) which can be adjusted in the future after some playtesting feedback indicates that this is too early/late.